src/archive.h

```diff
@@ -39,19 +36,86 @@ class QString;
 
 class FileData {
 public:
-  virtual QString getFileName() const = 0;
-  virtual void addOutputFileName(QString const &fileName) = 0;
-  virtual std::vector<QString> getAndClearOutputFileNames() = 0;
+
+  /**
+   * @return the full filepath to this entry.
+   */
+  virtual std::wstring getFileName() const = 0;
+
+  /**
+   * @return the size of this entry in bytes (uncompressed).
+   */
+  virtual uint64_t getSize() const = 0;
+
+  /**
+   * @brief Add the given filename to the list of files to create from this
+   *   entry when extracting.
+   *
+   * @param filepath The filename to add.
+   */
+  virtual void addOutputFileName(std::wstring const& filepath) = 0;
+  
+  /**
+   * @brief Retrieve the list of filepaths to extract this entry to and clear the
+   *   internal list.
+   *
+   * @return the list of paths this entry should be extracted to.
+   */
+  virtual std::vector<std::wstring> getAndClearOutputFileNames() = 0;
+  
+  /**
+   * @return the CRC of this file.
+   */
   virtual uint64_t getCRC() const = 0;
+
+  /**
+   * @return true if this entry is a directory, false otherwize.
+   */
   virtual bool isDirectory() const = 0;
 };
 
 
 class Archive {
+public: // Declarations
 
-public:
+  enum class LogLevel {
+    Debug,
+    Info,
+    Warning,
+    Error
+  };
+
+  enum class ProgressType {
+    
+    // Indicates the progression in the archive. This may reach 100% way before the

```

Comment:
What does this mean? The progress in reading the archive during the extraction? What should this be used for?

Reply:
Are you asking about the enumeration of about the `ARCHIVE` value? The enumeration is to select different progress type in `ProgressCallback`.

The `ARCHIVE` value is related to "reading" the archive during extraction, but I don't really now if it's `READING` (it's not clear at all in 7z "documentation"). It was what was used by MO2 prior to this PR to update progress, which is why for some archives, you either reached 99% very quickly and spent most of the time on the last %, or you reached 100% before the extraction was complete and then the progress dialog started disappearing / re-appearing in a loop until the extraction was complete.

I've let this because this can probably be useful sometimes. For instance, I may use this progress instead of `EXTRACTION` prior to installation because it might be more accurate in this case.

You'd have to test the PRs to see the difference.

Reply:
Yes I was asking what the archive flag meant vs extraction. From the comment it's not really clear if there is any use case for it, so I would add to that comment what you have found through testing, like the fact that it might be better for extracting just a few files.

Reply:
I've updated the comment for both enumeration values.